packages/grafana-ui/src/components/Combobox/MultiCombobox.tsx

```diff
@@ -40,9 +42,27 @@ export const MultiCombobox = <T extends string | number>(props: MultiComboboxPro
 
   const styles = useStyles2(getComboboxStyles);
 
-  const [items, _baseSetItems] = useState(isAsync ? [] : options);
+  const allOption = useMemo(() => {
+    return {
+      label: t('multicombobox.all.title', 'All'),
+      // Type casting needed to make this work when T is a number
+      value: 'all' as unknown as T,
+    };
+  }, []);
+
+  const [items, _baseSetItems] = useState(isAsync ? [] : canToggleAll ? [allOption, ...options] : options);
   const [isOpen, setIsOpen] = useState(false);
 
+  useEffect(() => {
+    if (!isAsync) {
+      if (canToggleAll) {
+        _baseSetItems([allOption, ...options]);
+      } else {
+        _baseSetItems(options);
```

Comment:
How does this scale? Would having just the `All` option show up as a pill be enough?

Reply:
I tried having it show up as something separate at the start that we could style how we wanted, but with `downshift` every option is set up with a bunch of things, so for instance to get it to work with the keyboard would be quite complicated. What is your worry about scaling?

Reply:
A couple of things:

- Performance, does it make sense to process all items (assuming multiple Comboboxes on a single page)
- Does the user understand that all values are selected?

Maybe we can skip looping through `selectedItems` when `all` is selected, and then just render a custom value pill?

Reply:
Ah I think I get you, you mean the pill that is shown on the actual Input? 
This part of the code is just setting the options in the menu though, so it is just adding an `All` option at the top if `canToggleAll` is true.
I think a pill in the input would be a question for Ed as well. I agree with the performance issue with that, but when a `MultiCombobox` is closed it doesn't actually show all items, only the ones that fit, the main issue is when you expand it, it then displays ALL the selected items.